webapp/src/components/common/Matrix/components/MatrixUpload/index.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2024, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+/**
+ * MatrixUpload.tsx - Main component that manages matrix uploads
+ */
+import { StudyMetadata } from "@/common/types";
+import DatabaseUploadDialog from "@/components/common/dialogs/DatabaseUploadDialog";
+import FileUploadDialog from "@/components/common/dialogs/FileUploadDialog";
+import { useTranslation } from "react-i18next";
+
+interface MatrixUploadProps {
+  studyId: StudyMetadata["id"];
+  path: string;
+  type: "file" | "database";
+  open: boolean;
+  onClose: VoidFunction;
+  onFileUpload?: (file: File) => Promise<void>;
+  fileOptions?: {
```

Comment:
Theses options should not be generic because it's a component only for Matrix

Reply:
ok will see that